polygon/sync/header_downloader.go

```diff
@@ -54,25 +52,32 @@ func newHeaderDownloader(
 	p2pService p2p.Service,
 	heimdall heimdall.HeimdallNoStore,
 	headersVerifier AccumulatedHeadersVerifier,
-	headersWriter HeadersWriter,
+	bodiesVerifier BodiesVerifier,
+	storage Storage,
 	notEnoughPeersBackOffDuration time.Duration,
 ) HeaderDownloader {
 	return &headerDownloader{
 		logger:                        logger,
 		p2pService:                    p2pService,
 		heimdall:                      heimdall,
 		headersVerifier:               headersVerifier,
-		headersWriter:                 headersWriter,
+		bodiesVerifier:                bodiesVerifier,
+		storage:                       storage,
 		notEnoughPeersBackOffDuration: notEnoughPeersBackOffDuration,
 	}
 }
 
+//
+// TODO (subsequent PRs) rename to block downloader? revise and update naming in logs, tests, etc.
```

Comment:
https://github.com/ledgerwatch/erigon/pull/9681 - chained to this PR (first merge current PR)